notifier.go

```diff
@@ -43,7 +43,8 @@ func (notifier *Notifier) Notify(err error, rawData ...interface{}) (e error) {
 			if config.Synchronous {
 				return (&payload{event, config}).deliver()
 			}
-			go (&payload{event, config}).deliver()
+			// Ensure that any errors are logged if they occur in a goroutine.
+			go (&payload{event, config}).deliverOrLog()
```

Comment:
Can we make this an anonymous function here so that we can still call notify in this code block?

```
go func(event *Event, config *Configuration) {
  err := (&payload{event, config}).deliver()
  if err != nil {
    config.log("bugsnag.Notify: %v", err)
  }
}(event, config)
```